### PR TITLE
Complement checks that consider HMI response to be invalid

### PR DESCRIPTION
**Pull request by dcherniev**
Added additional check for one of the cases which should consider received HMI response as invalid response.

### CODE REVIEW

**Comment by dcherniev** [comment]
Please review: @indlu, @VVeremjova, @LitvinenkoIra, @LuxoftAKutsan

File: src/components/application_manager/src/commands/command_request_impl.cc [hunk 1]
```diff
@@ -247,7 +246,20 @@ void CommandRequestImpl::SendResponse(
...
-  response[strings::msg_params][strings::result_code] = result_code;
+
+  // In case result type was not recognized the response that we received from
+  // HMI is considered as invalid response
+  if (mobile_apis::Result::INVALID_ENUM == result_code) {
+    LOG4CXX_ERROR(logger_,
+                  "HMI response contains unrecognizable result code, so it is "
+                  "considered as invalid response");
+    response[strings::msg_params][strings::result_code] =
+        hmi_apis::Common_Result::GENERIC_ERROR;
```
Comment (VVeremjova) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @dcherniev You send message to mobile, but send code from hmi_apis.

Comment (dcherniev) [inline src/components/application_manager/src/commands/command_request_impl.cc]: @VVeremjova You right, fixed in 2834639.

**Comment by dcherniev** [comment]
Rebuild required

**Comment by dcherniev** [comment]
Rebuild required

**Comment by iCollin** [comment]
Closed in favor of https://github.com/smartdevicelink/sdl_core/pull/3718

**Pull request closed without merging**